cmd/manta-factoryreset.js

```diff
@@ -523,6 +523,33 @@ async.waterfall(
             });
         },
 
+        function _removePoseidonDcLocalConfig(cb) {
+            if (!POSEIDON) {
+                cb();
+                return;
+            }
+
+            var ufds = self.UFDS;
+            var log = self.log;
+
+            ufds.listDcLocalConfig(POSEIDON.uuid, function(err, config) {
```

Comment:
I'm sure this change is fine, and there are other uses of ufds.deleteDcLocalConfig(..) that match this one, but I'm confused about how node function signatures work.

https://github.com/joyent/node-ufds/blob/master/lib/index.js#L1984 shows that we take four parameters:

```
function deleteDcLocalConfig(uuid, account, datacenter, cb) {
```

but that `account` is optional. How can we get away with calling this with only three parameters (POSEIDON.uuid, config.dclocalconfig, cb) - I was expecting the 2nd parameter (account) to be passed as 'null' or something?

Reply:
node-ufds is the worst display of argument handling that we have in our repos. Yes it is very confusing.


```javascript
function listDcLocalConfig(uuid, account, cb, noCache) {
    var base;
    if (typeof (account) === 'function') {
        noCache = cb;
        cb = account;
        account = null;
    }
    account = workAroundUnderspecification(account, 'account');

    if (!account) {
        base = sprintf(USER_FMT, uuid);
    } else {
        base = sprintf(SUBUSER_FMT, uuid, account);
    }
    cb = once(cb);

    // ...
}

listDcLocalConfig(<uuid>, <callback>);
```

This results in:

```
- arguments:
    uuid: <uuid>
    account: <callback>
    cb: undefined
    noCache: undefined
- if-block:
    noCache = undefined
    cb = <callback>
    account = null
    // uuid stays as <uuid>
- base = sprintf(USER_FMT, <uuid>)
```

Reply:
Ah! I hadn't grokked the 
```
    if (typeof (datacenter) === 'function') {
        cb = datacenter;
        datacenter = account;
        account = null;
    }
```
jees.  Thanks for the explanation!